package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import warnings
+
+from MDAnalysis.analysis.base import AnalysisBase
+
+def dihedral_calc(atomgroups):
+    """Calculates phi and psi angles for a list of AtomGroups over trajectory.
+
+    Parameters
+    ----------
+    atomgroups : list of AtomGroups
+        must be a list of one or more AtomGroups containing 5 atoms in the
+        correct order (i.e. C-N-CA-C-N)
+
+    Returns
+    -------
+    angles : numpy.ndarray
+        An array of time steps which contain (phi,psi) for all AtomGroups.
+    """
+
+    dihedrals = [atomgroup.dihedral for atomgroup in atomgroups]
+    angles = [dih.value() for dih in dihedrals]
+
+    return angles
+
+
+class Ramachandran(AnalysisBase):
+    """Calculate phi and psi dihedral angles of specified residues.
+
+    Note
+    ----
+    Run the analysis with :meth:`Ramachandran.run()`, which stores the results
+    in the array :attr:`Ramachandran.angles`. A axes object can be obtained
+    with :meth: `Ramachandran.run().plot()`.
+
+    If the residue selection is beyond the scope of the protein, then an error
+    will be raised. If the residue selection includes the first or last residue
+    then a warning will be raised, and the final array of angles will not
+    include those residues.
+
+    """
+    def __init__(self, atomgroup, **kwargs):
+        r"""Parameters
+        ----------
+        atomgroup : Atomgroup
+            atoms for residues for which phi and psi are calculated
+        start : int, optional
+            starting frame, default None becomes 0.
+        stop : int, optional
```

Comment:
aren't we moving those to the run function @orbeckst

Reply:
yes, you're right, of course – I asked @hfmull to start from analysis.rms.RMSD and forgot that it uses soon-to-be changed call signatures (#1463).

Do we need to keep init start/stop/step for new additions like this one and deprecate them  (#1979) right away?? I would say _no_ (don't teach the user something you want them to forget) but I am open to alternative opinions.

Reply:
still keeping them here makes this behave exactly as any other class. Kind of nice.

Reply:
It makes more sense to not allow them here if we want them in `run`